nativelib/src/main/resources/posix/sys/select.c

```diff
@@ -0,0 +1,95 @@
+#include <sys/select.h>
+#include <stdbool.h>
+#include <errno.h>
+#include <stddef.h>
+#include <string.h>
+
+struct scalanative_timeval {
+	time_t tv_sec;
+	suseconds_t tv_usec;
+};
+
+int scalanative_FD_SETSIZE() { return FD_SETSIZE; }
+
+void convert_scalanative_timeval(struct scalanative_timeval *in, struct timeval *out) {
+	out->tv_sec = in->tv_sec;
+	out->tv_usec = in->tv_usec;
+}
+
+void scalanative_FD_ZERO(int *set) {
+	memset(set, '\0', sizeof(*set));
+}
+
+void scalanative_FD_CLR(int fd, int *set) {
+	set[fd/sizeof(int)] &= ~((int)(1<<(fd % sizeof(int))));
+}
+
+void scalanative_FD_SET(int fd, int *set) {
+	set[fd/sizeof(int)] |= ((int)(1<<(fd % sizeof(int))));
+}
+
+int scalanative_FD_ISSET(int fd, int *set) {
+	return (set[fd/sizeof(int)] & ((int)(1<<(fd & sizeof(int)))));
+}
+
+int scalanative_select(int nfds, int *readfds, int *writefds,
+		       int *exceptfds, struct scalanative_timeval *scalanative_timeout) {
+	fd_set rfds;
```

Comment:
fd_set is different on all platforms. There is probably a better way to convert my implementation of fd_set to system's implementation though.